libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshApi.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.navigation.core.routerefresh
+
+import android.util.Log
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.api.directionsrefresh.v1.MapboxDirectionsRefresh
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRefreshResponse
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+data class RouteRefreshRequest(
+    val accessToken: String,
+    val originalRoute: DirectionsRoute,
+    val requestUuid: String,
+    val legIndex: Int
+)
+
+class RouteRefreshApi {
+
+    suspend fun refresh(
```

Comment:
I'm not excited by the way this handles error logging. But I do like how everything is off the ui thread (i think).

either way, reference to the legacy version which is still good
https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.kt

Reply:
updated to make error handling nice